### ISSUE TITLE
Fix the MarkerTransferAuthorization Accept function and TransferCoin authz handling to prevent problems when other authorization types are used

### ISSUE THREAD

**Issue by egaxhaj** [provenance-io/provenance#903]
## Summary of Bug

When there is a `MsgTransferRequest` the `Accept` method decrements funds and returns an updated `MarkerTrasnferAuthorization` with the new limit amount when there exists an `msg.administrator`. Then the funds are again decremented in the `authzHandler` function in the Marker module. This may lead to double decrementing when the transfer type is something other than `MsgTransferRequest`.

Line 710:
https://github.com/provenance-io/provenance/blob/1283582fdc559857ce2d5fc1af3a39a904abd471/x/marker/keeper/marker.go#L699-L717

We're deleting here when `limitLeft.IsZero()` and ignoring `accept.Delete` which is set by `shouldDelete` (see below). If we  check for `if accept.Delete { ... }` it helps avoid `type casting` and allows for other authorization types to work as well.

The logic in `authzHandler()` and `MarkerTransferAuthorization.Accept()` are confusing. 

https://github.com/provenance-io/provenance/blob/1283582fdc559857ce2d5fc1af3a39a904abd471/x/marker/types/authz.go#L26-L44

Line 40: Why is the update occurring in the `marker` module?

## Steps to Reproduce

____

## For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by dwedul-figure** [comment]
After some research, it looks like this is working as intended, except accidentally, and in such a way that would be easy to have things go wrong.

In the `Accept` method, the limit is decremented if the `from == admin`. In the `TransferCoin` endpoint, we only call `Accept` when `from != admin`. But then, in the `TransferCoin` endpoint, if `from != admin`, we then manually decrement the limit.

That decrementing should ONLY be handled in the `Accept` method. In the `TransferCoin` endpoint, where we call `Accept`, we then need to properly handle its result, i.e. deleting if appropriate, or updating with the returned authorization if appropriate.

Basically, the `MarkerTransferAuthorization.Accept` method should always return the updated authorization and delete flag (without the `from`/`admin` comparison). Then the `authzHandler` should not care what type of authorization is being used as long as `Accept` says it's okay. It should also use the result of `Accept` to decide when to update or delete the authorization entry.

**Comment by dwedul-figure** [comment]
I guess there **_is_** still actually an error.

Because `authzHandler` assumes that the authorization is a `MarkerTransferAuthorization` authorization, any other type of authorization will get messed up. E.g. if it's a generic authorization, `limitLeft, _ := authorization.(*types.MarkerTransferAuthorization).DecreaseTransferLimit(amount)` will always result in a negative limit (if it doesn't panic), which will then cause the authorization to be deleted.
